source/Calamari.Tests/KubernetesFixtures/Authentication/SetupKubectlAuthenticationFixture.cs

```diff
@@ -160,12 +161,12 @@ public void Execute_WhereNamespaceDoesNotExist_NamespaceCreated()
         {
             variables.Set(SpecialVariables.ClusterUrl, ClusterUrl);
             variables.Set(SpecialVariables.Namespace, Namespace);
-
+            variables.Set(Deployment.SpecialVariables.Account.AccountType, AccountTypes.UsernamePassword);
             invocations.FailFor("kubectl", $"get namespace {Namespace} --request-timeout=1m");
 
             var sut = CreateSut();
 
-            var result = sut.Execute(AccountTypes.UsernamePassword);
```

Comment:
Since we dont decide the auth mechanism on the account type alone (it also uses other variables) it seems wrong to pass the account type itself through the execute function.

Reply:
My personal preference would be to go the other way and front load more variable data, then pass some kind of `IKubectlAuthenticationParams` with an implementation for each account type. That way we'll do less plucking of variables mid method.

I had a brief look at how that might end up though and feels like a bit of a can of worms, so I'm leaving this comment as food for thought but I'm not convinced it would be a worthwhile change.

Reply:
This is the nature of the variable bag structure. What variables are needed depend entirely on the different auth mechanism. This makes it very clear that _everything_ is up for grabs :)